### PR TITLE
Multiline Input v2

### PR DESCRIPTION
**Pull request by libertymayc**
Includes integration with Form Field
New `question` prop on FormFieldLabel for question styling
cc @bhoppers2008

### CODE REVIEW

File: packages/core/stories/form-field/form-field.doc.stories.mdx [hunk 1]
```diff
@@ -28,12 +28,18 @@ All examples demonstrate Salt form controls inside the Form Field, including Inp
...
 ## Form Controls
 
-Form field can wrap any Salt UI control that is typically found in a form, such as Input, Checkbox and Radio Button components.
+Form field can wrap any Salt UI control that is typically found in a form, such as Input, Multiline Input, Checkbox and Radio Button components.
 
 <Canvas>
   <Story id="core-form-field--with-controls" />
 </Canvas>
 
+When using Form Field with Multiline Input, the `question` prop can be used on the form label and the `fullBorder` prop can be set on Multiline Input to style the field as a question.
```
Comment (libertymayc) [inline packages/core/stories/form-field/form-field.doc.stories.mdx:37]: I assume this isn't worded best - needs to be reworded properly in accordance with your design guidance @bhoppers2008

File: packages/core/src/form-field/FormFieldLabel.tsx [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
-import { useComponentCssInjection } from "@salt-ds/styles";
-
 import formFieldLabelCss from "./FormFieldLabel.css";
 
 const withBaseName = makePrefixer("saltFormFieldLabel");
 
 export const FormFieldLabel = ({
   className,
   children,
+  question = false,
```
Comment (alycrys) [inline packages/core/src/form-field/FormFieldLabel.tsx:15]: was a variant option discussed?

File: packages/core/stories/form-field/form-field.doc.stories.mdx [hunk 3]
```diff
@@ -62,6 +68,18 @@ The Form Field Label can be positioned to the left of the form control via the p
 
 Place your label to the left of the control if your application is limited in vertical real estate or if you have a very data heavy, dense form, where it’s preferable to have as many fields visible as possible without scrolling.
 
+## Form Field Label Question
+
+The Form Field Label can be styled as a question by setting the `question` prop to `true`.
```
Comment (alycrys) [inline packages/core/stories/form-field/form-field.doc.stories.mdx:73]: what does it mean to be styled as a question? I feel this is confusing. Can we have a variant=bold or something?

Comment (alycrys) [reply packages/core/stories/form-field/form-field.doc.stories.mdx:73]: I also think this is easily achievable with additional styling applied by the users. 
Should it be just an example?

Comment (libertymayc) [reply packages/core/stories/form-field/form-field.doc.stories.mdx:73]: it's going to be a common use case cc @bhoppers2008 - I would go against variant=bold as it goes against primary/secondary we have elsewhere

Comment (libertymayc) [reply packages/core/stories/form-field/form-field.doc.stories.mdx:73]: the wording can definitely be improved though

Comment (alycrys) [reply packages/core/stories/form-field/form-field.doc.stories.mdx:73]: Of course, you're right about the variant=bold (wasn't being serious about it tbh), but I would personally avoid `question`.

File: packages/core/src/form-field/FormFieldLabel.tsx [hunk 4]
```diff
@@ -1,20 +1,22 @@
...
-import { useComponentCssInjection } from "@salt-ds/styles";
-
 import formFieldLabelCss from "./FormFieldLabel.css";
 
 const withBaseName = makePrefixer("saltFormFieldLabel");
 
 export const FormFieldLabel = ({
   className,
   children,
+  pronounced = false,
```
Comment (joshwooding) [inline packages/core/src/form-field/FormFieldLabel.tsx:15]: Should pronounced be a boolean? Has design thought about if there would be an opposite to this

File: packages/lab/src/__tests__/__e2e__/multiline-input/MultilineInput.cy.tsx [hunk 5]
```diff
@@ -159,4 +159,83 @@ describe("GIVEN an MultilineInput", () => {
...
+  describe("WHEN used in Formfield", () => {
+    describe("AND disabled", () => {
+      it("THEN MultilineInput within should be disabled", () => {
+        cy.mount(
+          <FormField disabled>
+            <FormFieldLabel>Disabled form field</FormFieldLabel>
+            <MultilineInput defaultValue="Value" />
+          </FormField>
+        );
+        cy.wait(1000);
```
Comment (joshwooding) [inline packages/lab/src/__tests__/__e2e__/multiline-input/MultilineInput.cy.tsx:172]: Why are these waits needed?

File: packages/lab/stories/multiline-input/multiline-input.doc.stories.mdx [hunk 6]
```diff
@@ -88,6 +90,8 @@ You can set Multiline Input to a read-only state. The value text can’t be edit
   <Story id="lab-multiline-input--readonly" />
 </Canvas>
 
+- When setting Multiline Input to read-only state, ensure to also disable any nested interactive adornments, such as buttons.
```
Comment (joshwooding) [inline packages/lab/stories/multiline-input/multiline-input.doc.stories.mdx:93]: Is this always the case? How would we support a copy button?

File: packages/lab/stories/multiline-input/multiline-input.doc.stories.mdx [hunk 7]
```diff
@@ -72,6 +72,8 @@ You can set Multiline Input to a disabled state, with no resultant action when t
   <Story id="lab-multiline-input--disabled" />
 </Canvas>
 
+- When setting Multiline Input to disabled state, ensure to also disable any nested interactive adornments, such as buttons.
```
Comment (joshwooding) [inline packages/lab/stories/multiline-input/multiline-input.doc.stories.mdx:75]: This sentence sounds a bit clunky to me, but content can review this later

File: packages/core/src/input/Input.tsx [hunk 8]
```diff
@@ -199,7 +199,7 @@ export const Input = forwardRef<HTMLDivElement, InputProps>(function Input(
         tabIndex={isReadOnly || isDisabled ? -1 : 0}
         onBlur={handleBlur}
         onChange={handleChange}
-        onFocus={!isDisabled ? handleFocus : undefined}
+        onFocus={!isDisabled && !isReadOnly ? handleFocus : undefined}
```
[the remark this answers is not in the document]
Comment (joshwooding) [reply packages/core/src/input/Input.tsx:202]: I asked about this, apparently the feature came from an accessibility SME

[the remark this answers is not in the document]
Comment (origami-z) [reply packages/core/src/input/Input.tsx:202]: Can we put an inline comment about this? Not SEM bit, the actual reasoning

File: packages/core/src/form-field/FormFieldLabel.tsx (hunk 4 above)
Comment (Fercas123) [reply packages/core/src/form-field/FormFieldLabel.tsx:15]: @bhoppers2008

File: packages/lab/stories/multiline-input/multiline-input.doc.stories.mdx (hunk 7 above)
Comment (Fercas123) [reply packages/lab/stories/multiline-input/multiline-input.doc.stories.mdx:75]: cc. @sion-docs / @mikearildbrown

File: packages/core/src/input/Input.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (origami-z) [reply packages/core/src/input/Input.tsx:202]: Not just screen-reader we should consider, keybaord user would not be able to navigate by tab and copy?

[the remark this answers is not in the document]
Comment (ZarkoPernar) [reply packages/core/src/input/Input.tsx:202]: Yeah, that what I was thinking of. Tabbing through form fields is very nice UX for all users. Besides, this is different to how the native input + readonly behaves in a browser.

[the remark this answers is not in the document]
Comment (joshwooding) [reply packages/core/src/input/Input.tsx:202]: A pattern for copying could easily be to add a Button next to the input, but in my mind, we should just build what has already been agreed.

File: packages/core/stories/form-field/form-field.stories.tsx [hunk 9]
```diff
@@ -131,8 +137,19 @@ export const LabelLeft: ComponentStory<typeof FormField> = (props) => {
   );
 };
 
+export const LabelQuestion: ComponentStory<typeof FormField> = (props) => {
+  return (
+    <FormField {...props}>
+      <FormLabel pronounced>
+        Do your current qualifications align with the role? Please describe.
+      </FormLabel>
+      <MultilineInput fullBorder defaultValue="Answer" />
```
[the remark this answers is not in the document]
Comment (alycrys) [reply packages/core/stories/form-field/form-field.stories.tsx:146]: so fullBorder stays then?

[the remark this answers is not in the document]
Comment (joshwooding) [reply packages/core/stories/form-field/form-field.stories.tsx:146]: I spoke to @Fercas123 yesterday about it, I think we should change it to either border or bordered. My preference is bordered.

[the remark this answers is not in the document]
Comment (alycrys) [reply packages/core/stories/form-field/form-field.stories.tsx:146]: How do you feel about pronounced?

**Review by alycrys — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**